### PR TITLE
Support lambda event v2

### PR DESCRIPTION
**Pull request by xinghengwang**
https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -234,73 +234,165 @@ module.exports = function (options, handler) {
...
-      'logEvent',
-      'AWS Lambda trigger must be a Load Balancer or API Gateway. ' +
-        'See https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-set-up-simple-proxy.html#api-gateway-set-up-lambda-proxy-integration-on-proxy-resource'
-    );
-    return Promise.resolve();
+// in V2
+// https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html
+// if there is no statusCode but valid JSON, API gateway will interpret
+// the response.
+function translateLambdaResultIfNeeded(result, isV1) {
```
Comment (xinghengwang) [inline lib/index.js:245]: @keyur9 I decided to support the "v2" result interpretation that 
if valid json or string, but no statusCode, API gateway will put the result as body, and assume status code of 200. 

https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html

Comment (xinghengwang) [reply lib/index.js:245]: specifically this text: 
"With the 2.0 format version, API Gateway can infer the response format for you. API Gateway makes the following assumptions if your Lambda function returns valid JSON and doesn't return a statusCode:"

File: lib/index.js [hunk 2]
```diff
@@ -234,73 +234,165 @@ module.exports = function (options, handler) {
...
   }
 
-  if (options.skip(event, context)) {
-    // exit early
-    return Promise.resolve();
+  if (result && result.statusCode) {
+    return result;
   }
 
+  if (!result.statusCode && (typeof result === 'object' || typeof result === 'string')) {
```
Comment (dgilling) [inline lib/index.js]: Is it possible to return a null/undefined result in Lambda?

**Review by dgilling — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 2 above)
Comment (xinghengwang) [reply lib/index.js]: I assume so.

**Review by dgilling — APPROVED** [review]

**Pull request merged**